### PR TITLE
Move custom ratelimit impl to common controller

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Purpose
fixes: https://github.com/wso2/apk/issues/1444

### CODE REVIEW

File: common-controller/internal/operator/controller/ratelimitpolicy_controller.go [hunk 1]
```diff
@@ -136,14 +136,17 @@ func (ratelimitReconsiler *RateLimitPolicyReconciler) Reconcile(ctx context.Cont
 		return ctrl.Result{}, nil
 	}
 	logger.Info("add api ratelimit")
-	var vhost, resolveRatelimit = ratelimitReconsiler.marshelRateLimit(ctx, ratelimitKey, ratelimitPolicy)
+	var vhost, resolveRatelimit, customRateLimitPolicy = ratelimitReconsiler.marshelRateLimit(ctx, ratelimitKey, ratelimitPolicy)
```
Comment (Krishanx92) [inline common-controller/internal/operator/controller/ratelimitpolicy_controller.go:139]: shall we move this to new method for keep the code clean

File: common-controller/internal/operator/controller/ratelimitpolicy_controller.go [hunk 2]
```diff
@@ -136,14 +136,17 @@ func (ratelimitReconsiler *RateLimitPolicyReconciler) Reconcile(ctx context.Cont
...
+	logger.Info("resolveCustomRateLimitPolicy", customRateLimitPolicy)
+
+	// if vhost == nil {
+	// 	logger.Info("vhost is nil hence API deleted")
+	// 	return ctrl.Result{}, nil
+	// }
+
 	ratelimitReconsiler.ods.AddorUpdateRatelimitToStore(ratelimitKey, resolveRatelimit)
-	xds.UpdateRateLimitXDSCache(vhost, resolveRatelimit)
+	xds.UpdateRateLimitXDSCache(vhost, resolveRatelimit, customRateLimitPolicy)
```
Comment (Krishanx92) [inline common-controller/internal/operator/controller/ratelimitpolicy_controller.go:149]: Need to handle the delete path as well

File: common-controller/internal/operator/controller/ratelimitpolicy_controller.go [hunk 3]
```diff
@@ -136,14 +136,17 @@ func (ratelimitReconsiler *RateLimitPolicyReconciler) Reconcile(ctx context.Cont
...
-	var vhost, resolveRatelimit = ratelimitReconsiler.marshelRateLimit(ctx, ratelimitKey, ratelimitPolicy)
+	var vhost, resolveRatelimit, customRateLimitPolicy = ratelimitReconsiler.marshelRateLimit(ctx, ratelimitKey, ratelimitPolicy)
 	logger.Info("resolveRatelimitxxxx", resolveRatelimit)
-	if vhost == nil {
-		logger.Info("vhost is nil hence API deleted")
-		return ctrl.Result{}, nil
-	}
+	logger.Info("resolveCustomRateLimitPolicy", customRateLimitPolicy)
+
+	// if vhost == nil {
```
Comment (Krishanx92) [inline common-controller/internal/operator/controller/ratelimitpolicy_controller.go:143]: Any reason to remove this?

File: common-controller/internal/operator/controller/ratelimitpolicy_controller.go (hunk 1 above)
Comment (CrowleyRajapakse) [reply common-controller/internal/operator/controller/ratelimitpolicy_controller.go:139]: ack

File: common-controller/internal/operator/controller/ratelimitpolicy_controller.go (hunk 2 above)
Comment (CrowleyRajapakse) [reply common-controller/internal/operator/controller/ratelimitpolicy_controller.go:149]: ack

File: common-controller/internal/operator/controller/ratelimitpolicy_controller.go (hunk 3 above)
Comment (Krishanx92) [reply common-controller/internal/operator/controller/ratelimitpolicy_controller.go:143]: we need to check vhost && !customRatelimit then continue, becasuse if we host is null and flow continue api ratelimit will braek

File: common-controller/internal/operator/controller/ratelimitpolicy_controller.go (hunk 1 above)
Comment (CrowleyRajapakse) [reply common-controller/internal/operator/controller/ratelimitpolicy_controller.go]: fixed

File: common-controller/internal/operator/controller/ratelimitpolicy_controller.go (hunk 2 above)
Comment (CrowleyRajapakse) [reply common-controller/internal/operator/controller/ratelimitpolicy_controller.go:158]: fixed

File: common-controller/internal/operator/controller/ratelimitpolicy_controller.go (hunk 3 above)
Comment (CrowleyRajapakse) [reply common-controller/internal/operator/controller/ratelimitpolicy_controller.go]: fixed

**Comment by CrowleyRajapakse** [comment]
Tested flows: Screenshot 2023-07-24 at 12 07 16

**Review by Krishanx92 — APPROVED** [review]

**Pull request merged**